week3/src/components/Button.jsx

```diff
@@ -0,0 +1,119 @@
+import { useEffect, useState } from "react";
+import styled from "styled-components";
+import { easyCardList } from "../constants/cardImgList";
+import { normalCardList } from "../constants/cardImgList";
+import { hardCardList } from "../constants/cardImgList";
+import CommonPage from "./CommonPage";
+
+const Button = () => {
+  const levelData = ["EASY", "NORMAL", "HARD"];
+  // 기본 난이도 'EASY'로 설정
+  const [isClick, setIsClick] = useState(0);
+  const [currentPage, setCurrentPage] = useState(<CommonPage />);
+  const [shuffle, setShuffle] = useState(0);
+
+  useEffect(() => {
+    switch (isClick) {
+      case 0:
+        setCurrentPage(<CommonPage cardList={easyCardList} />);
+        break;
+
+      case 1:
+        setCurrentPage(<CommonPage cardList={normalCardList} />);
+        break;
+
+      case 2:
+        setCurrentPage(<CommonPage cardList={hardCardList} />);
+        break;
+
+      default:
+        break;
+    }
```

Comment:
요거 isClick을 요기서 switch문으로 나누지 말고 요렇게 바로 넣고, 공통적인 함수로 수행할 수 있도록 리팩토링 할 수도 있을 것 같아! ㅎㅎ 밑에 mixCardList부분에도 달아놨당!!

```suggestion
 setCurrentPage(<CommonPage cardList={mixCardList(isClick)} />);
```